test/integration/api/datasets.js

```diff
@@ -349,6 +349,24 @@ describe('datasets and entities', () => {
                 .catch(error => {
                   error.constraint.should.be.equal('check_blobId_or_datasetId_is_null');
                 }))))));
+
+      it('should throw problem if datasetId is being set for non-data type', testService((service, { Forms, FormAttachments, Datasets }) =>
```

Comment:
modify this test to call PATCH attachment

Reply:
Didn't modify this test, instead added two additional tests to ensure that dataset can only be linked to attachments with type='file'